pennylane_qiskit/qiskit_device2.py

```diff
@@ -98,7 +103,27 @@ def circuit(x):
     """
     # Code to acquire session:
     existing_session = device._session
-    session = Session(backend=device.backend)
+
+    # When an existing session exists, we want to use its settings unless overwritten
+    # by settings in the qiskit_session
```

Comment:
I still consider the behaviour quite unintuitive. If we want to go for this, can we include information in the docstring about what will happen if the device that is passed already has a session on it?

Reply:
No I think it is kind of unintuitive to do it this way as well.

The best way is probably to just make the docstring more clear that if you initialize `qiskit_session` you will not be using your existing session set in the device.

The logic is that I think the existing session set in device should only have value in a very specific case: batched circuits. In all other cases, the session is just necessary to access the primitives but doesn't necessarily _do_ anything. 

Therefore if a user has gone out of their way to use `qiskit_session`, they probably want to do some kind of like VQE or QAOA or whatever iterative thing, so they probably want those settings. I'll make it clearer that a User **doesn't** accidentally think they should set the Session in the device.